src/bindings/python/src/pyopenvino/frontend/tensorflow/decoder_base.hpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <pybind11/pybind11.h>
+#include "openvino/frontend/tensorflow/decoder.hpp"
+
+namespace py = pybind11;
+
+/// Trampoline class to support inheritance from GraphIterator in Python
+class PyDecoderBase : public ov::frontend::tensorflow::DecoderBase {
+
+    ov::Any get_attribute(const std::string &name) const override{
+        PYBIND11_OVERRIDE_PURE(ov::Any, DecoderBase, get_attribute, name);
+    }
+
+    size_t get_input_size() const override{
+        PYBIND11_OVERRIDE_PURE(size_t, DecoderBase, get_input_size);
+    }
+
+
+    std::string get_input_node_name(size_t input_port_idx) const {
+        PYBIND11_OVERRIDE_PURE(std::string, DecoderBase, get_input_node_name, input_port_idx);
+    }
+
+    size_t get_input_node_name_output_port_index(size_t input_port_idx) const {
+        PYBIND11_OVERRIDE_PURE(size_t, DecoderBase, get_input_node_name_output_port_index, input_port_idx);
+    }
+
+    std::string get_input_node_name_output_port_name(size_t input_port_idx) const {
+        PYBIND11_OVERRIDE_PURE(std::string, DecoderBase, get_input_node_name_output_port_name, input_port_idx);
+    }
+
+    void get_input_node(size_t input_port_idx,
+                                std::string &producer_name,
+                                std::string &producer_output_port_name,
+                                size_t &producer_output_port_index) const override{
+        producer_name = get_input_node_name(input_port_idx);
+        producer_output_port_index = get_input_node_name_output_port_index(input_port_idx);
+        producer_output_port_name = get_input_node_name_output_port_name(input_port_idx);
+    }
```

Comment:
can we implement this also in Python and remove auxiliary methods `get_input_node_name_output_port_index`, `get_input_node_name_output_port_name` here?

Reply:
No, the problem is with input/output parameters (`producer_name`, `producer_output_port_name`, `producer_output_port_index`). 
Basic types like `str` or `int`. Such types are always passed to methods by value, not by reference. 
For example:
```
def get_input_node(input_port_idx: int, producer_name: str, producer_output_port_name: str, producer_output_port_index: int):
      producer_name = some_value
      producer_output_port_name = some_value
      producer_output_port_index = some_value


producer_name = old_value
get_input_node(input_port_idx, producer_name, producer_output_port_name, producer_output_port_index)
print(producer_name) # has old value
```

The only way to make such parameters be passed by reference is to wrap them to some complex class (list or tuple for example) or use some helper methods, which I did. 
So we cannot create direct binding for get_input_node(). It is a limitation from PyBind explained by Python specifics.
https://pybind11.readthedocs.io/en/stable/faq.html#limitations-involving-reference-arguments